shell/platform/android/android_shell_holder.cc

```diff
@@ -129,13 +158,19 @@ void AndroidShellHolder::Launch(RunConfiguration config) {
 
   shell_->GetTaskRunners().GetUITaskRunner()->PostTask(
       fxl::MakeCopyable([engine = shell_->GetEngine(),  //
-                         config = std::move(config)     //
+                         config = std::move(config),    //
+                         view = platform_view_.get()    //
   ]() mutable {
-        if (engine) {
-          if (!engine->Run(std::move(config))) {
-            FXL_LOG(ERROR) << "Could not launch engine in configuration.";
-          }
+        bool success = false;
+        FXL_LOG(INFO) << "Attempting to launch engine configuration...";
+        if (!engine || !engine->Run(std::move(config))) {
+          FXL_LOG(ERROR) << "Could not launch engine in configuration.";
+        } else {
+          FXL_LOG(INFO) << "Isolate for engine configuration successfully "
+                           "started and run.";
+          success = true;
         }
+        view->InvokeOnStartedCallback(success);
```

Comment:
Attach a closure to the `root_isolate_create_callback` field of the settings object used to start this shell. This will fire the callback only if the isolate is correctly launched and also take care of re-firing the callback in case of cold-restart events (which the plugins also probably need to react to). You will lose the ability for the plugin to react to failed launches. But, then again, there isn't much it can do now either. So, we may have to devise a more robust error recovery/reporting mechanism. But till that time, lets use the existing hooks.

Be careful that the callback in the settings object is fire on the UI thread, you might have to post a task back to the platform thread to fire a message back to the platform.

Reply:
On second thought, I don't think this is particularly useful if we have no way of notifying the plugin that we failed to start. I've gone ahead and removed this.